libraries/lib-time-and-pitch/StaffPad/TimeAndPitch.cpp

```diff
@@ -403,7 +419,11 @@ void TimeAndPitch::setTimeStretchAndPitchFactor(double timeScale, double pitchFa
 {
   assert(timeScale > 0.0);
   assert(pitchFactor > 0.0);
-  _pitchFactor = pitchFactor;
+  if (_pitchFactor != pitchFactor)
+  {
+    _pitchFactor = pitchFactor;
+    setupInputBuffers();
```

Comment:
I haven't tested this library with dynamic pitch or time ratio changes, but I think `setTimeStretchAndPitchFactor` is supposed to be callable on the fly. `setupInputBuffers` seem to reset memory, which I suppose would now introduce glitches in such use case.

Reply:
> I think setTimeStretchAndPitchFactor is supposed to be callable on the fly

Agree; this would be needed when we implement "tempo track"